tiledb/sm/cpp_api/array.h

```diff
@@ -177,16 +280,35 @@ class Array {
   /**
    * Compute an upper bound on the buffer elements needed to read a subarray.
    *
+   * **Example:**
+   * @code{.cpp}
+   * tiledb::Context ctx;
+   * tiledb::Array array(ctx, "s3://bucket-name/array-name", TILEDB_READ);
+   * std::vector<int> subarray = {0, 2, 0, 2};
+   * auto max_elements = array.max_buffer_elements(subarray);
+   *
+   * // For fixed-sized attributes, `.second` is the max number of elements
+   * // that can be read for the attribute. Use it to size the vector.
+   * std::vector<int> data_a1(max_sizes["a1"].second);
+   *
+   * // In sparse reads, coords are also fixed-sized attributes.
+   * std::vector<int> coords(max_sizes[TILEDB_COORDS].second);
+   *
+   * // In variable attributes, e.g. std::string type, need two buffers,
+   * // one for offsets and one for cell data.
+   * std::vector<uint64_t> offsets_a1(max_sizes["a2"].first);
+   * std::vector<char> data_a1(max_sizes["a2"].second);
+   * @endcode
+   *
    * @tparam T The domain datatype
-   * @param uri The array URI.
-   * @param schema The array schema
    * @param subarray Targeted subarray.
-   * @return The maximum number of elements for each array attribute (plus
-   *     coordinates for sparse arrays).
-   *     Note that two numbers are returned per attribute. The first
-   *     is the maximum number of elements in the offset buffer
-   *     (0 for fixed-sized attributes and coordinates),
-   *     and the second is the maximum number of elements of the value buffer.
+   * @return A map of attribute name (and `TILEDB_COORDS` for sparse arrays) to
```

Comment:
`TILEDB_COORDS` is not just for sparse arrays. It is also for dense, as the user now can ask explicitly for the coordinates.